tests/ops/op_math/test_controlled.py

```diff
@@ -808,8 +838,7 @@ def test_control_values_special_decomp(self, test_expand):
 
         decomp = op.expand().circuit if test_expand else op.decomposition()
         expected = [qml.PauliX(1), qml.Toffoli((0, 1, 2)), qml.PauliX(1)]
-        for op1, op2 in zip(decomp, expected):
-            assert qml.equal(op1, op2)
+        assert equal_if_decomposed(decomp, expected)
```

Comment:
If the decomposition has now changed, we should changed `expected` and continue to use `qml.equal`.